examples/arm/arm-cross.nix

```diff
@@ -0,0 +1,96 @@
+let crossPkgs = pkgs.pkgsCross.aarch64-multiplatform;
+    haskellNix = import (builtins.fetchTarball https://github.com/input-output-hk/haskell.nix/archive/b406bb1d5bed581f651ae18a5d3ed07f47ace2b9.tar.gz) {};
+    pkgs = import haskellNix.sources.nixpkgs haskellNix.nixpkgsArgs;
+    iserv-proxy = pkgs.buildPackages.ghc-extra-packages.ghc8102.iserv-proxy.components.exes.iserv-proxy;
+    remote-iserv = crossPkgs.ghc-extra-packages.ghc8102.remote-iserv.components.exes.remote-iserv;
+    llvm = pkgs.llvm_9;
+    crossNumactl = crossPkgs.numactl;
+    qemu = pkgs.buildPackages.qemu;
+    qemuIservWrapper = pkgs.writeScriptBin "iserv-wrapper" ''
+      #!${pkgs.stdenv.shell}
+      set -euo pipefail
+      # Unset configure flags as configure should have run already
+      unset configureFlags
+      PORT=$((5000 + $RANDOM % 5000))
```

Comment:
This looks like it's prone to flaky build errors if this happens to land on an occupied port. Does iserv perhaps support port=0 and a port file or something of the sort to safely choose a free port?

Reply:
Here is where the binding happens,
https://gitlab.haskell.org/ghc/ghc/-/blob/master/libraries/libiserv/src/Remote/Slave.hs#L64
We can pass 0 to remote iserv, and it will bind to an available port, but it won't report back which port it was.

We are forced then to do something hacky. One easy way would be to try to start remote-iserv a few times before giving up.

Reply:
>  We can pass 0 to remote iserv, and it will bind to an available port, but it won't report back which port it was.

I see, that's unfortunate. Perhaps worth a ticket on iserv.

> We are forced then to do something hacky. One easy way would be to try to start remote-iserv a few times on different random ports before giving up.

:+1:

Reply:
I made a try. Instead of starting remote-iserv multiple times, I'm using `nc` to find an available port. It turns out to be simpler if it works.

Reply:
One more try, this time without using nc. I think the approach using ss is less invasive as it doesn't need to interact with services running on occupied ports.